src/Core/Services/Implementations/OrganizationService.cs

```diff
@@ -2150,6 +2152,50 @@ private async Task ValidateOrganizationCustomPermissionsEnabledAsync(Guid organi
         }
     }
 
+    private async Task ValidateCustomPermissions(OrganizationUser user)
```

Comment:
This method is only relevant if `user.Type == OrganizationUserType.Custom`, so it should either have an early return in this case, or only be called when required. (I personally prefer only calling methods when they're needed)

Reply:
The method is now called on `ValidateOrganizationUserUpdatePermissions` when `newType == OrganizationUserType.Custom`